modin/core/dataframe/pandas/dataframe/dataframe.py

```diff
@@ -2192,7 +2192,7 @@ def apply_full_axis(
         apply_indices=None,
         enumerate_partitions: bool = False,
         dtypes=None,
-        keep_partitioning=True,
+        num_splits="keep",
```

Comment:
I don't think it is good to have `num_splits` parameter, which can take different values like str (meaningful) or int. We have `num_splits` parameter that is just int in many places. These parameters may be mixed up at some point. Should we leave `keep_partitioning` as is, add a new parameter called `num_splits`, and introduce the logic of handling `keep_partitioning` and `num_splits` at the Dataframe layer?

Reply:
Sounds reasonable.

Made changes so now `num_splits` could only be an integer and the combination of `keep_partitioning` and `num_splits` parameters is handled at the partition mgr level.